Doc/c-api/datetime.rst

```diff
@@ -106,6 +106,12 @@ Macros to create objects:
    .. versionadded:: 3.6
 
 
+.. c:function:: PyObject* PyTime_FromTime(int hour, int minute, int second, int usecond)
```

Comment:
Why do you move this paragraph?

Reply:
I moved that paragraph to create a better order as requested by @pganssle so that PyTime_FromTimeAndFold follows PyTime_FromTime which is a better order for a user who will be reading the docs.